### DISCUSSION TITLE
Polling a device for battery level with the CCAPI

### DISCUSSION THREAD

**Discussion by jmbaute** [zwave-js/node-red-contrib-zwave-js#289] [Q&A] [answered]
I am using a simple temp/humidity sensor and have been successful with captuing the messages when those values change. I am attempting to use the CMD Factory to trigger the device to send out the battery level. So far, I haven't seen it respond to the message I'm sending in (I can see and capture the battery level message if I remove/reinsert the battery from the device). I am hoping someone can help me configure the CMD factory node to send the proper message. I've looked over the docs here and the zwave-js documentation and I'm just not getting something. Here is my relevant flow:
![image](https://github.com/zwave-js/node-red-contrib-zwave-js/assets/220179/8dabf955-512a-48de-9db5-0b9436636845)

The node is configured as follows
![image](https://github.com/zwave-js/node-red-contrib-zwave-js/assets/220179/604f1836-710f-44bc-b60e-b0905d93d8d9)

The message being sent is
![image](https://github.com/zwave-js/node-red-contrib-zwave-js/assets/220179/7b110bc7-07f5-438b-aec5-41db226a73c1)

Thanks to anyone for help!!

**Comment by marcus-j-davies** [comment, marked as answer]
Hi @jmbaute,

What you have is correct.

But the clue is in the fact it's a battery operated device.
It is not awake to receive your request  to report the battery - and in fact may not be needed at all...

You see, battery operated devices employ a "wake up" interval, and in doing so, will automatically report the battery when they wake up in that interval.

Can you change it? - sure, as long as it does actually have a wake interval, see below.

![Screenshot 2023-12-13 at 20 00 24](https://github.com/zwave-js/node-red-contrib-zwave-js/assets/55892693/3991c446-1549-4fef-8a8b-b05cf4cda2f4)

Battery operated devices  are not "listening" for requests - and will only awake on the interval I mentioned or when they are triggered to do so, i.e a change in temp, motion etc etc

Any requests you have sent will be queued, and processed when the device checks in on its interval

You can just use the Value API if you wanted to get the battery level as it was last reported.

I don't recommend the CCAPI - the Value API is far superior.

**Reply by jmbaute** [reply]
Hi @marcus-j-davies . Thanks for the quick reply! Your explanation makes sense. I just happened to see the wake/sleep cycle in my debugger, but it didn't report anything - I'm guessing because the battery is at 100%. 

With the value API, does that ask the controller for the last value that was reported to it? Or will there be a response from the device I have to listen for? In my applicaiton it is important to have a periodic report of the battery level, but could go weeks or maybe a month.

Note: I am not able to change any of the property values of my sensor in the API. I press "set" but it goes back to whatever was there.

**Comment by marcus-j-davies** [comment]
> I'm guessing because the battery is at 100%.

Quite likely, the manual of the device, should offer more explanation, some devices have a config setting also (Configuration CC), that may control this aspect - its really down to the manufacture what they do with reporting the battery.

> With the value API, does that ask the controller for the last value that was reported to it? Or will there be a response from the device I have to listen for? In my application it is important to have a periodic report of the battery level, but could go weeks or maybe a month.

The Value API stores ALL values that the device will report, and you can fetch these values at anytime, (as they are stored) - The CCAPI - is extremely low level, so does not offer any type of caching.

The Value API data cache is resistant to reboots, as its written to disc when the values are updated.
The UI uses the Value API to display current values - i.e the battery for example

![Screenshot 2023-12-13 at 20 19 41](https://github.com/zwave-js/node-red-contrib-zwave-js/assets/55892693/aa12aaea-d54b-4088-a395-e0a67ed77dad)


> Note: I am not able to change any of the property values of my sensor in the API. I press "set" but it goes back to whatever was there.

The changes are queued, and will get applied, when the device awakes.
Some devices have a wake up button, so you could "hurry" the change, if not - all values that you set will get applied when the device awakes -this is just the nature of battery operated devices.

**Reply by jmbaute** [reply]
Thanks so much - and, I got the value working!

```
{
  "networkId": 1,
  "node": 2,
  "nodeName": "Temp1",
  "nodeLocation": "",
  "event": "GET_VALUE_RESPONSE",
  "timestamp": 1702498753759,
  "object": {
    "commandClassName": "Battery",
    "commandClass": 128,
    "endpoint": 0,
    "property": "level",
    "propertyName": "level",
    "currentValue": 100,
    "normalizedObject": {
      "commandClass": "128 - Battery",
      "type": "number",
      "currentValue": 100,
      "label": "Battery level",
      "unit": "%"
    }
  }
}
```

**Reply by marcus-j-davies** [reply]
Nice - in theory, I would hope the device will report its battery - but nothing stopping you once a day to issue a get using the CCAPI, at least when it awakes, it will be forced to update us (as the Value API will be notified of the new value).

remember to review your wake up interval - as that will help - but don't set a silly value if you want your battery to last, and remember the new interval, won't be set until it awakes up to retrieve the new setting

Note : I formatted  your payload for easier reading
